server/threadEvent/queries.ts

```diff
@@ -1,50 +1,61 @@
 import { ThreadEvent, attributesPublicUser } from 'server/models';
+import * as types from 'types';
+import { ThreadEventPermissions } from './permissions';
 
-export const createThreadEvent = (inputValues, userData) => {
-	return ThreadEvent.create({
+export const createThreadEvent = async (
+	inputValues: {
+		threadId: string;
+		type: string;
+		data: Record<string, any>;
+	},
+	userData: types.User,
+) => {
+	const newThreadEvent = await ThreadEvent.create({
 		type: inputValues.type,
 		data: inputValues.data,
 		userId: userData.id,
 		threadId: inputValues.threadId,
-	}).then((newThreadEvent) => {
-		/* Populate user data so it can be inserted into */
-		/* existing pubData client-side */
-		const cleanedUserData = {};
-		attributesPublicUser.forEach((key) => {
-			cleanedUserData[key] = userData[key];
-		});
-		return {
-			...newThreadEvent.toJSON(),
-			user: cleanedUserData,
-		};
 	});
+	/* Populate user data so it can be inserted into */
+	/* existing pubData client-side */
+	const cleanedUserData = {};
+	attributesPublicUser.forEach((key) => {
+		cleanedUserData[key] = userData[key];
+	});
+	return {
+		...newThreadEvent.toJSON(),
+		user: cleanedUserData,
+	};
 };
 
-export const updateThreadEvent = (inputValues, updatePermissions) => {
+export const updateThreadEvent = async (
+	inputValues: types.ThreadEvent & { threadEventId: string },
+	updatePermissions: ThreadEventPermissions['update'],
+) => {
 	// Filter to only allow certain fields to be updated
 	const filteredValues = {};
 	Object.keys(inputValues).forEach((key) => {
-		if (updatePermissions.includes(key)) {
+		// @ts-expect-error FIXME: updatePermissions can only be `false`, this shouldn't be here
+		if (updatePermissions && updatePermissions?.includes(key)) {
```

Comment:
`updatePermissions` can only be `false`, so the part after `&&` is `never`, which cannot have `includes`.

In general this pattern is badly typed (filteredValues is always `{}`), but I restrained myself from messing with it too much haha